src/modules/category/useCase/DeleteCategory.ts

```diff
@@ -0,0 +1,20 @@
+import { HttpException } from '../../../errors/HttpException';
+import { CategoryRepository } from '../repository/CategoryRepository';
+
+export class DeleteCategory {
+  private repository;
+
+  constructor(repository: CategoryRepository) {
+    this.repository = repository;
+  }
+
+  async execute(id: number, id_student: string): Promise<void> {
+    const category = await this.repository.findOneById(id, id_student);
+
+    if (category?.id_student !== id_student) {
```

Comment:
Como o método "findOneById" busca somente valores que contenham exatamente os valores de "id" e "id_student ", e o retorno traz um objeto com seus valores ou null, você pode simplificar a verificação com esta mudança.

```suggestion
    if (!category) {
```

Reply:
Resolvido